README.md

```diff
@@ -440,6 +440,15 @@ The creation of the `--help` option can be disabled by passing `null` as the
 constructing a `HelpFormatter` instance. In the above example a
 `DefaultHelpFormatter` was created with the prologue and epilogue.
 
+For correct show help message you must add handler for help exception:
+```
+val parser = try {
+    ArgParser(args).parseInfo(::YourParser)
+} catch (e: ShowHelpException) {
+    e.printAndExit()
+}
+```
+
```

Comment:
Thanks for pointing that this was not clear enough!

The supported way to do this is actually to use `mainBody`, or at least to catch all `SystemExitException`s, as mentioned in the Error Handling section.

Perhaps just change this to say something like:

> For help messages to correctly display you must properly handle any `SystemExitException` that is thrown as described in **Error Handling**.

You could even link to that section.

Reply:
Thank you for your response.

I think, that catch only awaited exception (such as `ShowHelpException`) is best way. In this case we need to catch only exception which will say to show help and catching `ShowHelpException` will give us all what we need.

Reply:
I don't think I understand your point.

`ShowHelpException` is a subclass of `SystemExitException`. In most cases you probably want to handle all `SystemExitException`s, not just `ShowHelpException`. They all contain a message for the user, as well as an exit code. The exit code for `ShowHelpException` just happens to be 0.

Reply:
I mean that any other exception must be handled by other way. In this case we need to catch just `ShowHelpException` and it is will be ok for show help message

Reply:
But you don't need to handle other SystemExitExceptions in a different way. All SystemExitExceptions can be handled by calling their `printAndExit()` method or by using `mainBody` (which will call `printAndExit` for you). It would be pretty unusual to need to catch only `ShowHelpException`.